File: webapi/Classes/Ticket.cs
using System;
using System.ComponentModel.DataAnnotations;
using Stripe;

namespace webapi.Classes
{
    public class Ticket
    {
        [Key] [Required] public Guid Id { get; set; }
        public string UserId { get; set; }
        public ApplicationUser User { get; set; }
        public Guid EventSeatId { get; set; }
        public EventSeat EventSeat { get; set; }
        public bool IsValid { get; set; }
        public DateTime ValidUntil { get; set; }
        public InvalidationReason? InvalidationReason { get; set; }
    }
}

File: angularapp/src/interfaces/ticket.ts

import {EventSeat} from "./event-seat";
import {InvalidationReason} from "./invalidation-reason";

export interface Ticket {
  id: string;
  userId: string;
  eventSeatId: string;
  eventSeat: EventSeat;
  isValid: boolean;
  validUntil: string;
  invalidationReason: InvalidationReason | null;
}
